tests/test_metadata_serialization.py

```diff
@@ -29,9 +32,87 @@
 logger = logging.getLogger(__name__)
 
 
+# pylint: disable=too-many-public-methods
 class TestSerialization(unittest.TestCase):
     """Test serialization for all classes in 'tuf/api/metadata.py'."""
 
+    invalid_metadata: utils.DataSet = {
+        "no signatures field": '{"signed": \
+            { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+            "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}} \
+        }',
+        "non unique duplicating signatures": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [ \
+                { \
+                    "keyid": "id", \
+                    "sig": "abc" \
+                }, \
+                { \
+                    "keyid": "id", \
+                    "sig": "abc" \
+                } \
+              ] \
+        }',
+    }
+
+    @utils.run_sub_tests_with_dataset(invalid_metadata)
+    def test_invalid_metadata_serialization(self, test_case_data: str) -> None:
+        case_dict = json.loads(test_case_data)
+        with self.assertRaises((ValueError, KeyError)):
+            Metadata.from_dict(case_dict)
+
+    valid_metadata: utils.DataSet = {
+        "multiple signatures": '{ \
+            "signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [ \
+                { \
+                    "keyid": "59a4df8af818e9ed7abe0764c0b47b4240952aa0d179b5b78346c470ac30278d", \
+                    "sig": "5b00100e9cf1c083f8347371ab840cf60124780305124ed7a53fe31bf43473c90b1d2c802ed2f11f5057ba21e6b7a05118b1907f737d2e29c9692aa3345f9801" \
+                }, \
+                { \
+                    "keyid": "8a1c4a3ac2d515dec982ba9910c5fd79b91ae57f625b9cff25d06bf0a61c1758", \
+                    "sig": "de0e16920f87bf5500cc65736488ac17e09788cce808f6a4e85eb9e4e478a312b4c1a2d7723af56f7bfb1df533c67d8c93b6f49d39eabe7fae391a08e1f72f01" \
+                } \
+            ] \
```

Comment:
opinions on making test data smaller in this way?

```suggestion
            "signatures": [ { "keyid": "a", "sig": "b" }, { "keyid": "c", "sig": "d" } ] \
```

I think the massive test data hides what the test does -- it looks like the keyids and signatures are important for the test when they are not. From that perspective the test data should be minimal to test the case that we want to test.

Reply:
You said you want me to use real signatures:
> this is also really valid_metadatas. I think valid_signatures (with actual signatures in it) might be more useful
from https://github.com/theupdateframework/python-tuf/pull/1809#discussion_r797441847?

Reply:
so I did... I think I may have been wrong then because A) the serialization does not test the signatures themselves -- just whether the signature object is valid and B) 11 lines for the signatures looks so unreadable

My bad, sorry. Either way works for me.

Reply:
I prefer if they are shorter, so yea, I will update it.

Reply:
Wait, I take that back.

I said _"valid_signatures (with actual signatures in it) might be more useful"_. This still sounds reasonable: I probably would not have objected to that.

But your test case here is **valid_metadata**. Adding 22 lines of unused signature data into the Metadata test (but not doing that in the signature test) is not as useful as it hides what is actually tested

Reply:
Well, you can have a look at my latest changes and tell me how you finally want it :D.
If it's still doesn't look good I will rever my changes and add the valid signatures inside the signatures test.